### ISSUE TITLE
[Support]: Long time to load clips

### ISSUE THREAD

**Issue by rayzorben** [blakeblackshear/frigate#6018]
### Describe the problem you are having

When viewing a clip, it takes 15-17 seconds for a clip to load to start playing. 

I am running in HassOS Addon on an RPI4 that boots off an NVME drive, but I cant be certain if hardware acceleration is working if that is possibly the cause.

### Version

Frigate (Full Access) Beta (0.12.0)

### Frigate config file

```yaml
ffmpeg:
  hwaccel_args: -c:v h264_v4l2m2m
  
detect:
  width: 704
  height: 480
  fps: 5

snapshots:
  enabled: True
    
record:
  enabled: True
  events:
    retain:
      default: 10
        
birdseye:
  enabled: True
  mode: continuous

objects:
  filters:
    person:
      min_area: 800
    car:
      max_area: 80000
  track:
    - car
    - person
    - dog

cameras:
```

### Relevant log output

```shell
2023-04-12 17:32:09.233334073  [INFO] Starting Frigate...
2023-04-12 17:32:13.891652169  [2023-04-12 10:32:13] frigate.app                    INFO    : Starting Frigate (0.12.0-bc16ad1)
2023-04-12 17:32:14.048788631  [2023-04-12 10:32:14] frigate.app                    INFO    : Creating directory: /tmp/cache
2023-04-12 17:32:14.057153102  [2023-04-12 10:32:14] peewee_migrate                 INFO    : Starting migrations
2023-04-12 17:32:14.072146255  [2023-04-12 10:32:14] peewee_migrate                 INFO    : There is nothing to migrate
2023-04-12 17:32:14.167817572  [2023-04-12 10:32:14] frigate.app                    INFO    : Output process started: 301
2023-04-12 17:32:14.171847254  [2023-04-12 10:32:14] detector.coral                 INFO    : Starting detection process: 300
2023-04-12 17:32:17.547970535  [2023-04-12 10:32:14] frigate.app                    INFO    : Camera processor started for front_porch: 306
2023-04-12 17:32:17.549043668  [2023-04-12 10:32:14] frigate.app                    INFO    : Camera processor started for driveway: 309
2023-04-12 17:32:17.549836841  [2023-04-12 10:32:14] frigate.app                    INFO    : Camera processor started for south_side: 311
2023-04-12 17:32:17.550657533  [2023-04-12 10:32:14] frigate.app                    INFO    : Camera processor started for back_north: 313
2023-04-12 17:32:17.551482188  [2023-04-12 10:32:14] frigate.comms.dispatcher       INFO    : Turning off recordings for front_porch
2023-04-12 17:32:17.552698503  [2023-04-12 10:32:14] frigate.app                    INFO    : Camera processor started for garage: 315
2023-04-12 17:32:17.553725026  [2023-04-12 10:32:14] frigate.app                    INFO    : Capture process started for front_porch: 316
2023-04-12 17:32:17.554868305  [2023-04-12 10:32:14] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2023-04-12 17:32:17.556756574  [2023-04-12 10:32:14] frigate.comms.dispatcher       INFO    : Turning off recordings for back_north
2023-04-12 17:32:17.557662524  [2023-04-12 10:32:14] frigate.app                    INFO    : Capture process started for driveway: 318
2023-04-12 17:32:17.559041449  [2023-04-12 10:32:14] frigate.app                    INFO    : Capture process started for south_side: 322
2023-04-12 17:32:17.565387153  [2023-04-12 10:32:14] frigate.app                    INFO    : Capture process started for back_north: 329
2023-04-12 17:32:17.567383772  [2023-04-12 10:32:14] frigate.app                    INFO    : Capture process started for garage: 341
2023-04-12 17:32:17.583579648  [2023-04-12 10:32:17] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
```

### FFprobe output from your camera

```shell
NA
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

### Network connection

Wired

### Camera make and model

Amcrest

**Comment by NickM-27** [comment]
you will want to use the network tools in your browser to see which call is taking time when loading the clip. We usually see this when a user has recordings on a NAS and has not put the DB on their local storage but it sounds like this is not your case

**Comment by rayzorben** [comment]
Looks like `seg-1-v1.ts`. Some of them only took 7 seconds, but this one took far more.

![image](https://user-images.githubusercontent.com/7737834/231556020-1d115f49-79f9-48cf-806e-30b286a2a23e.png)

**Comment by NickM-27** [comment]
so it's either having trouble the metadata for the segment or the CPU is being slow to handle that work.Do you have an ffprobe for that camera?

**Comment by rayzorben** [comment]
I don't know how to get the ffprobe from HassOs addon, this is what I get from the ffprobe button in System:

[{"return_code":1,"stderr":"","stdout":""},{"return_code":1,"stderr":"","stdout":""}]

**Comment by NickM-27** [comment]
okay, that is odd. are the cameras h264 or h265?

**Comment by rayzorben** [comment]
![image](https://user-images.githubusercontent.com/7737834/231561209-514f76bd-a114-4363-b373-7dd49a257476.png)

**Comment by rayzorben** [comment]
Is there anything I can do to help debug? I got host access to hassos now, so I can probably log into the docker container and get any information?

**Comment by NickM-27** [comment]
I'm not really sure, curious what @blakeblackshear as I have not seen cases where the delay is on the segment itself (usually is master.m3u8)

**Comment by rayzorben** [comment]
Is the recording saved on disk somewhere or in the database?

**Comment by NickM-27** [comment]
it is saved on disk under /media/frigate/recordings

**Comment by blakeblackshear** [comment]
If you navigate away and then come back to play the same recording, is it fast the second time around?

**Comment by rayzorben** [comment]
> If you navigate away and then come back to play the same recording, is it fast the second time around?

If I close and reopen the 'tab' it is still slow. If I navigate back to Camera and back again and expand it, it is still slow.

I don't have the greatest upload speed, 10MBit, so doing the math it would take about 17 seconds at that speed to download the whole file. But are you sending the whole file to start playing it? I didn't think so as I played a clip that is 35 minutes long and it takes the same 20ish seconds.

**Comment by NickM-27** [comment]
what is the latency of the network? that may explain part of it along with the slow upload speed. It is HLS so the whole clip is not loaded at once, but latency will mean longer delays between each call.

**Comment by blakeblackshear** [comment]
If it takes a long time on the second request, the only thing I can think of is that you are saturating the I/O bandwidth of the Pi. I would recommend turning off services that write to the NVMe and see if performance improves. Turn off record temporarily, turn off other addons that write to the SSD, etc.

**Comment by rayzorben** [comment]
I don't think that is it, like I mentioned, when I attempt to play it from VLC across SMB it loads in under 3 seconds.

 I already have almost nothing running but Frigate and HA. But I turned off everything I can to verify and it is still slow to load.

![image](https://user-images.githubusercontent.com/7737834/232099558-d6845927-140d-4ead-8ae1-4433fea3f10c.png)

![image](https://user-images.githubusercontent.com/7737834/232100511-a3c28940-eeba-4550-9824-361fac7456b0.png)


I checked and see little to no activity on the disk at all

```
# iostat -d 2
Linux 5.15.84-v8 (homeassistant)        04/14/23        _aarch64_       (4 CPU)

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.03         0.00       4752          8
zram1             0.00         0.00         0.02        632       3360
zram2             0.23         0.00         1.84        752     290808
sda               8.25        17.29       104.37    2739165   16534756

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda              21.50         0.00       204.00          0        408

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda               4.00         0.00        36.00          0         72

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda              35.00         0.00       312.00          0        624

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda               0.00         0.00         0.00          0          0

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda               0.00         0.00         0.00          0          0

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda              18.50         0.00       164.00          0        328

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda               0.00         0.00         0.00          0          0

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda               0.00         0.00         0.00          0          0

Device:            tps   Blk_read/s   Blk_wrtn/s   Blk_read   Blk_wrtn
zram0             0.00         0.00         0.00          0          0
zram1             0.00         0.00         0.00          0          0
zram2             0.00         0.00         0.00          0          0
sda              17.00         0.00       148.00          0        296

# hdparm -t /dev/sda

/dev/sda:
Timing buffered disk reads:   71 MB in 3.03 seconds = 23927 kB/s
# hdparm -T /dev/sda

/dev/sda:
Timing buffer-cache reads:   668 MB in 0.51 seconds = 1339697 kB/s
#
```

What is odd, is that a 21 second video takes exactly 21 seconds to load and play. But a 20 second video takes only 8 seconds:

![image](https://user-images.githubusercontent.com/7737834/232099494-0805eb5a-4832-450d-a45b-11355a33f450.png)

**Comment by rayzorben** [comment]
Is there anything I can do to help debug? I have no problem giving you access to the host/HA

**Comment by rayzorben** [comment]
Pinging again, not trying to be a bother. Is there some tracing I can do? Can I see the source for seg-1-v1.ts? See where it is spending the most time?

**Comment by NickM-27** [comment]
the source is in nginx vod module, not sure there's much that can be enabled there. Besides an I/O issue, not sure what it would be. Frigate recordings are stored in 10 second segments so a 21 second event (3 segments) taking longer to load than a 20 second event (2 segments) seems to support that.

**Comment by NickM-27** [comment]
> How do you suggest I prove or disprove the IO issue or bandwidth issue? If VLC can play it in under 3 seconds, it suggests that it is neither.

What URL are you using in VLC to test?

> Is this the project for that module? https://github.com/kaltura/nginx-vod-module I can also open an issue there if you think it would be helpful.

yes that's the project, I don't think it's a bug but if they have support issues then I'm sure they could provide more information to debug or potential explanations.

**Comment by NickM-27** [comment]
So you're playing the mp4 file directly? Because that is not a valid comparison at all compared to playing the vod in inginx

**Comment by rayzorben** [comment]
> So you're playing the mp4 file directly? Because that is not a valid comparison at all compared to playing the vod in nginx

I am playing it across the internet via SAMBA. Which means it reads it from disk and sends it over the internet. 

I'd be glad to try another method that bypasses vod in nginx to prove or disprove any io or bandwidth issue.

**Comment by NickM-27** [comment]
Right, but that means it's just reading a single segment and playing it in VLC, not all 3. nginx reads all segments to get metatdata about them, then provides the vod in HLS with the files in .ts format (the file sent depends on if the source is h.264 or h.265). That's a lot of work that VLC is not doing.

You could try playing the master.m3u8 in VLC and see what happens.

**Comment by rayzorben** [comment]
> You could try playing the master.m3u8 in VLC and see what happens.

I'm not sure what that is, how do I get that?

**Comment by NickM-27** [comment]
you can get the url from the network tools or just get an event id and create the api url https://docs.frigate.video/integrations/api#get-vodeventevent-idindexm3u8

**Comment by rayzorben** [comment]
I have no idea what URL I should be using with running inside of HA. From the docs I tried this http://ccab4aaf_frigate-fa-beta:5000/api/front_porch but it does not seem to work.

**Comment by NickM-27** [comment]
It's in your screenshot, all you have to do is click on the file and it will show the full link 
![232099494-0805eb5a-4832-450d-a45b-11355a33f450](https://user-images.githubusercontent.com/14866235/234362401-3a666050-7618-463c-b96b-fe9f8cefbe95.png)

**Comment by rayzorben** [comment]
Thanks, that is what I did and it is not working

![image](https://user-images.githubusercontent.com/7737834/234364014-da2e0521-d19e-42a3-8c0d-e7b4363a57d4.png)

**Comment by NickM-27** [comment]
probably because ingress is being used, it would need to be done with the ip:5000 directly

**Comment by rayzorben** [comment]
The IP of HA? Should it be HTTP or HTTPS? I'm not even using VLC, just my web browser first to get the API URL correct.

I can reach https://192.168.1.10:8123 no issues
I cannot connect to http://192.168.1.10:5000, https://192.168.1.10:5000, or any specific URL such as /api/stats

**Comment by NickM-27** [comment]
right, because you need to enable the port in addon settings

**Comment by rayzorben** [comment]
Thanks, I did not know that existed!

So now that I have it, the video loads instantly in VLC. It starts playing right away but then pauses and continues about 1 second every second.

However, so does Frigate/VOD just with smaller pauses. It plays about 4-5 seconds, then buffers for a few and then continues. All in all it takes 57 seconds to 'download' the entire video

![image](https://user-images.githubusercontent.com/7737834/234381061-2a24af3e-0270-4e61-8649-87992d857467.png)

Why does this take so long when I can copy the whole file in 11.5 seconds (over the internet, the same connection) which requires both that it read it from disk fully and send it across the network.

```
PS X:\frigate\recordings\2023-04-16\22\front_porch> Measure-Command { cp .\19.58.mp4 c:\tmp }

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 11
Milliseconds      : 620
```

**Comment by NickM-27** [comment]
Interesting, no idea why that would be the case and it would only be guessing to say why. I ran a test for recordings that are stored on my SSD (first image) and then stored on my HDD with the HDD spun down. 

![Screen Shot 2023-04-25 at 13 33 56 PM](https://user-images.githubusercontent.com/14866235/234384774-e6898ff8-31cb-45bc-a0da-a700ec024dba.png)

![Screen Shot 2023-04-25 at 13 38 18 PM](https://user-images.githubusercontent.com/14866235/234384933-bba70319-9db1-4ea6-99b6-b8aa902469df.png)

**Comment by rayzorben** [comment]
The file is read from disk fully in 300 milliseconds (here is a non-cached file)

```
# dd if=./07.48.mp4 of=/dev/zero bs=512k count=2048
14+1 records in
14+1 records out
7690210 bytes (7.3MB) copied, 0.304828 seconds, 24.1MB/s
```

I really don't think that IO is the issue, all tests show all files reading super fast.

Is there any configuration for VOD that I can test?

I found a video that is only a single segment and I am testing with that. It takes 14 seconds which is close to the file copy time. Is there a way to split it into smaller chunks?

**Comment by NickM-27** [comment]
> I found a video that is only a single segment and I am testing with that. It takes 14 seconds which is close to the file copy time. Is there a way to split it into smaller chunks?

not sure what you mean exactly by splitting into smaller chunks but that would require re-encoding the file

**Comment by rayzorben** [comment]
Makes sense.

I am at a complete loss. I thought I had an AHA moment. With the IO or disk not being an issue, I started focusing on bandwidth as the potential problem, so I called my ISP. They mentioned that my upload was only 5mbit! It was supposed to be 10mbit. However I asked them what upload speed I can get, and they offered me 30mbit for a $5 upgrade. Sure. Verified it takes effect immediately, which they confirmed.

Then I installed speedtest addon for HA to verify and indeed I am at 30mbit now (previous tests were 5mbit+)

![image](https://user-images.githubusercontent.com/7737834/234407005-0bc2a669-fdec-4c35-b848-764bbda6038f.png)

So I got excited to see the clip load faster. This is the same one as last time, and NOPE still slow even though my upload is now 6x higher. One open of it even took LONGER than before!

![image](https://user-images.githubusercontent.com/7737834/234406848-5331e380-148b-4322-92ad-fadc6b58651a.png)

I really don't know what to do here unless there is a bug in nginx vod. I might try to pull an older version of frigate in a separate docker and see if anything works.

**Comment by NickM-27** [comment]
Interesting, though testing on http:// locally the upload speed wouldn't affect it. It any case, frigate 0.12 is almost at ~44,000 pulls so if this issue was a bug in the software itself I'd fully expect it to be much more widespread and others joining into this conversation as well.

**Comment by rayzorben** [comment]
Restored back to 11.1, same issue :(

**Comment by rayzorben** [comment]
> Interesting, though testing on http:// locally the upload speed wouldn't affect it. It any case, frigate 0.12 is almost at ~44,000 pulls so if this issue was a bug in the software itself I'd fully expect it to be much more widespread and others joining into this conversation as well.

Just to note, there is no 'locally'. This server runs at my moms house and I only have access over the internet via HA or through vpn (wireguard). Ping times are 33 ms, and bandwidth is 30mbit.

**Comment by rayzorben** [comment]
Me too. I should be back there in the month I can test it out if I don't figure it out by then

**Comment by rayzorben** [comment]
What settings/resolution is your recording stream?

![image](https://user-images.githubusercontent.com/7737834/234703882-9b76a405-c33c-4c29-a036-2aedb83e91f7.png)

**Comment by NickM-27** [comment]
Right now all my cameras are 2560x1920, 20fps, h.264, 6144 kbps

**Comment by rayzorben** [comment]
Shoot I was hoping that was the difference. I tried turning down the frame rate to 15fps and it didn't seem to solve it though :(

**Comment by rayzorben** [comment]
Sorry for bothering you, but when you can't find what's wrong, I tend to try to find what's the difference. 

Do you run on an rpi4 (mine is rpi4 4GB)? How does yours perform over the internet?

**Comment by NickM-27** [comment]
> Do you run on an rpi4 (mine is rpi4 4GB)? How does yours perform over the internet?

No, mine runs on a full NAS server that I built. It also performs very well over internet, but then again I have fiber internet with symmetrical gig service

**Comment by rayzorben** [comment]
So I think I have figured something out, let me know if you disagree and that I may be doing something wrong.

I modified `nginx.conf` and changed the `vod_hls_mpegts_interleave_frames` from `on` to `off`. I then did `nginx -s reload`.

As you can see, the first 18 seconds in that list was with it `on`. the next one is with it `off`. every call after that are different clips with it `off` and it consistently loads in 2 seconds and streams even smoother than before.

![image](https://user-images.githubusercontent.com/7737834/234989244-8f8ecce4-6f6d-41d7-9020-96a958a82f53.png)

From the documentation the default is off, so is there possibly a reason it is on?

```
vod_hls_mpegts_interleave_frames
syntax: vod_hls_mpegts_interleave_frames on/off
default: off
context: http, server, location
When enabled, the HLS muxer interleaves frames of different streams (audio / video). When disabled, on every switch between audio / video the muxer flushes the MPEG TS packet.
```

**Comment by NickM-27** [comment]
There's a few things here:
1. this value was set by me in updates to optimize nginx in 0.12. I believe you said previously that this problem was present in 0.11 as well.
2. This option is recommended to be set to true by nginx to improve performance https://github.com/kaltura/nginx-vod-module/blob/master/README.md#performance-recommendations

**Comment by rayzorben** [comment]
You are probably right. I seem to have a lot of variability going on which is making testing difficult.

**Comment by rayzorben** [comment]
Do you know much about nginx-vod? I tried enabling performance counters in nginx.conf to try to see where the time is being spent.

```
vod_performance_counters perf_counters;

location /vodstatus/ {
  vod_status;
  access_log off;
}
```

It works, it adds them to the vod status page:

![image](https://user-images.githubusercontent.com/7737834/235001239-ce2b4696-33c3-45a6-91b5-8eddfaf9a44b.png)


However I cannot play videos anymore, I get a weird error:

![image](https://user-images.githubusercontent.com/7737834/235001345-6eb3115f-e8bf-46ae-9e99-04ba40c52378.png)

**Comment by NickM-27** [comment]
not sure why that would affect things if that was the only change that was made

**Comment by rayzorben** [comment]
Yeah that is the only change. I comment out the performance counter line and leave the location there and everything works but once I uncomment the performance counter line I get that error when trying to play the video. I posted on the VOD NGINX module page so hopefully I can get some ideas from that.

**Comment by rayzorben** [comment]
comment on nginx-vod-module says it might have been a bug that was released in 2022 https://github.com/kaltura/nginx-vod-module/issues/1436#issuecomment-1528782912

It looks like frigate is using nginx-vod-module version 1.11 which seems to be released in 2016, am I reading that wrong?

![image](https://user-images.githubusercontent.com/7737834/235310834-485426a8-e798-4e56-bcc0-4bec8b0a201b.png)

**Comment by rayzorben** [comment]
Nevermind, that looks like a potential bug with build/nginx-vod-module as that is the default version. I see you are pulling 1.30 which is right before the ARM fix for perf counters. I'll just have to wait for a new version of frigate with 1.31+ to test it out.

**Comment by rayzorben** [comment]
@NickM-27 i'd hate to bring up this old topic again, but I finally got a new computer, x86_64 and I am having the same issue. So I was using the basis that I can play Plex movies just fine, but then I realized those are all 1080p. They start playing almost instantly.

When I tried a 4K movie though it took a very long time to get started. My cameras are 4K, so I am thinking maybe that is it?

Is there a way to play clips from the UI in 1080P? transcode?

**Comment by NickM-27** [comment]
I'd check the database and where it is stored, strikes me as some kind of issue with that. 

Might want to run VACUUM on the db

**Comment by rayzorben** [comment]
Absolutely a brand new setup, I didn't migrate anything from the previous
system just in case it was an issue with that.

**Comment by NickM-27** [comment]
I'd suggest making a new issue just so can recheck everything, but to answer your question no transcoding playback is not supported.

**Comment by kirsch33** [comment]
just as another data point, I revised `vod_hls_mpegts_interleave_frames` to `off` as well and saw a drastic decrease in recording load times

is this something we could dive more into on a new issue @NickM-27  or is there other factors that necessitate this option defaulting to `on`?

**Comment by NickM-27** [comment]
@kirsch33 the nginx vod module docs specifically recommend enabling it to improve performance for hls. During 0.12 I did a lot of testing and found that it improved performance for multiple test hosts / clients

**Comment by kirsch33** [comment]
interesting, theres got to be more to the story based on what im seeing personally, ill try to do some digging and open a new issue if this change remains consistently better in my case

**Comment by kirsch33** [comment]
@NickM-27 sorry if im beating a dead horse here, but after more testing I did confirm that this doesnt happen locally.

but over WAN, i have the same issue as @rayzorben with that `seg-1-v1.ts` taking quite a long time to load clips. 

is it simply being limited by my upload bandwidth or is there something else at play?

**Comment by rayzorben** [comment]
I just don't buy that it is a limited upload bandwidth issue.

I can copy the file over the internet from the network share it resides on (I have copied it to backup just to make sure it is the right file but it is the same drive/computer/etc). It takes 9.5 seconds to copy the WHOLE 11MB file

![image](https://github.com/blakeblackshear/frigate/assets/7737834/b7455498-7993-41cd-b149-bb8fd8462e94)

However, when playing through the web interface, it takes about 9.5 seconds just for the first segment, then 10.6 seconds for the 2nd segment, and finally 3 seconds for the final segment

<img width="465" alt="image" src="https://github.com/blakeblackshear/frigate/assets/7737834/94648b99-4259-43e2-9f3b-496d5d4f31ba">

**Comment by Yohann-SK** [comment]
I see the topic is closed, but I have same issue and do not understand how did you solve it. I am a new user of Frigate on home assistant, and everything is running really fast and great apart opening clips and records what takes around 20s. 
I am running frigate version: 0.12.1 on a beelink S12pro VM on proxmox. thx in advance for support.

**Comment by 0xfeedbac** [comment]
@rayzorben or @kirsch33 Can you please specify where is the nginx config that you edit? Is it some nginx that is built in the Frigate docker, or do you setup the reverse proxy as in this instruction? https://docs.frigate.video/guides/reverse_proxy/

**Comment by 0xfeedbac** [comment]
Can confirm the behavior. With `vod_hls_mpegts_interleave_frames off`, very long (>20 seconds) waiting times disappear, events from low resolution recording cameras start to play almost instantly, events from high resolution cameras - take longer to load, but nowhere near as it took. 

A quick reference for those who decide to reproduce the behavior (docker frigate installation): 
`docker ps -a`
`docker exec -it CONTAINER_ID bash `
in container, install text editor with `apt-get update && apt-get install nano` for instance
current default config location: 
`/usr/local/nginx/conf/nginx.conf`
`vod_hls_mpegts_interleave_frames off;`
`nginx -s reload`

**Comment by GrumpyMeow** [comment]
On my setup the playback of recordings always took a very long-time. But after making the change described by "fe-ed-ba-c" the playback starts immediatly. Previously an initial request would take 16seconds to transfer +/- 10mb. After that first segment, the video would start playing and the next segments got transferred.

**Comment by rayzorben** [comment]
@NickM-27 @blakeblackshear I really hate to be a bother, but this still plagues me and it seems others as well. I have even completely redone my system from what was a raspberry pi to an intel based system. When I am at home on the LAN clips work fine, but outside of the LAN it takes forever to load! Yet at the same time I can stream movies fine from PleX from the same server.

I haven't tested the change mentioned above again, but I kindly ask for some support on this?

**Comment by NickM-27** [comment]
I am not sure what can be done as it is now, none of the maintainers see this issue. We would need fresh data like was provided before so we can see what exactly is taking so much time.

It is worth noting that we have dropped video.js in the upcoming frigate 0.14 in favor of using standard html5 player with hls.js as a backup if the browser does not support hls. In my experience this has significantly reduced buffering times and playing the full recordings at 16x does not buffer at all where as with video.js it buffered every few seconds. So it is possible that these changes will fix this issue.

**Comment by NickM-27** [comment]
> If you are changing fundamentally how it is streamed then it isn't worth investing time in this.

let's be clear from the backend perspective it is streamed exactly the same, the change is purely to do with the frontend. However, data is requested differently by these frontend components so that is why I believe things may improve for you and other users seeing this.

**Comment by rayzorben** [comment]
That is clear. Let me know if I could be involved early on with .14 I could easily run it in parallel with .13 (I even have a spare USB Coral).

Thanks

**Comment by NickM-27** [comment]
you don't need permission to use 0.14, builds are done automatically https://github.com/users/blakeblackshear/packages/container/package/frigate

**Comment by nhickster** [comment]
I'm still seeing this issue when using Frigate 0.14.1-f4f3cfa. Clips take ~30 seconds to load while the live feed loads instantly. I tried the `vod_hls_mpegts_interleave_frames off` fix, but it still takes a long time for clips to load even after changing that.

I just have single reolink doorbell camera streaming at 2560*1920. Not sure if others are still having this issue, but just wanted to chime in and let you know that at least 1 person (me) is still reproducing it with the latest Frigate.

Do you want any debug info from me, or should I open a new issue?

**Comment by nhickster** [comment]
Here's the network output from firefox. I marked the section where I see the video still loading and when I finally see it start playing:
![image](https://github.com/user-attachments/assets/e59db61b-f8bb-4341-bdc6-0779ab8950b0)

**Comment by rayzorben** [comment]
I've pretty much given up on using frigate for actually watching the historical videos. I am literally on the same network (LAN) as my server, and I just can't watch them. Like you, live loads instantly. But I click on an event and it either never loads, or takes 10-15 seconds to load.

**Comment by Fahmula** [comment]
I'm experiencing the same issue. The vod_hls_mpegts_interleave_frames off option didn't help in my case. Playback doesn't start until seg-1-v1-a1.ts is fully downloaded, and it pauses when transitioning to the next segment.

![Screenshot 2024-12-31 205916](https://github.com/user-attachments/assets/ebcab3b2-8d62-4851-8af5-3f32a2b02156)

**Comment by jrcichra** [comment]
I'm having the same issue and by using as many default settings as possible I'm able to get VODs to load instantly now on my local network.

```nginx
    server {
        include listen.conf;

        # vod settings
        vod_base_url '';
        vod_segments_base_url '';
        vod_mode mapped;
        vod_max_mapping_response_size 1m;
        vod_upstream_location /api;
        vod_ignore_edit_list on;
```

I'm still seeing an issue when I route through a Cloudflare Tunnel that limits the VOD downloads to ~200KB/s. Even if I try to pre-cache a VOD ts by playing it back locally first.

**Comment by lpaolini** [comment]
I'm having the same problem. When I'm home (wired or wifi) both live feed and events load quickly.
However, when I'm on WireGuard over 5G cellular network (bandwidth 100Mbps+), live feed loads quickly, while events take forever or don't load at all.

I really don't understand what is the difference between the two cases.

**Comment by JPCodeCraft** [comment]
This still happens. I'm on 0.15.2 on Home assistant. Basically unusable outside local network for viewing recorded footage.
